examples/src/main/java/org/apache/ignite/examples/datagrid/store/CacheLoadOnlyStoreExample.java

```diff
@@ -0,0 +1,192 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.examples.datagrid.store;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+import javax.cache.configuration.FactoryBuilder;
+import javax.cache.integration.CacheLoaderException;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.Ignition;
+import org.apache.ignite.cache.CacheMode;
+import org.apache.ignite.cache.CachePeekMode;
+import org.apache.ignite.cache.store.CacheLoadOnlyStoreAdapter;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.internal.util.typedef.T2;
+import org.apache.ignite.lang.IgniteBiTuple;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Example of how to load data from CSV file.
+ *
+ * Note that it aims to illustrate the use of {@link CacheLoadOnlyStoreAdapter}
+ * and does not contains thorough error checking and validation.
+ */
+public class CacheLoadOnlyStoreExample {
+    /** Cache name. */
+    private static final String CACHE_NAME = CacheLoadOnlyStoreExample.class.getSimpleName();
+
+    public static void main(String[] args) throws Exception {
+        try (Ignite ignite = Ignition.start("examples/config/example-ignite.xml")) {
+            System.out.println();
+            System.out.println(">>> CacheLoadOnlyStoreExample started.");
+
+            ProductLoader productLoader = new ProductLoader("product.csv");
+
+            productLoader.setThreadsCount(2);
+            productLoader.setBatchSize(10);
+            productLoader.setBatchQueueSize(1);
+
+            try (IgniteCache<String, Product> cache = ignite.getOrCreateCache(cacheConfiguration(productLoader))) {
+                // load data.
+                cache.loadCache(null);
+
+                System.out.println(">>> Loaded number of items: " + cache.size(CachePeekMode.PRIMARY));
+
+                System.out.println(">>> Product id1's data: " + cache.get("id1"));
+            }
+            finally {
+                // Distributed cache could be removed from cluster only by #destroyCache() call.
+                ignite.destroyCache(CACHE_NAME);
+            }
+        }
+    }
+
+    /**
+     * Creates cache configurations for the loader.
+     *
+     * @return {@link CacheConfiguration}.
+     */
+    private static CacheConfiguration cacheConfiguration(ProductLoader productLoader) {
+        CacheConfiguration cacheCfg = new CacheConfiguration();
+
+        cacheCfg.setCacheMode(CacheMode.PARTITIONED);
+        cacheCfg.setName(CACHE_NAME);
+
+        cacheCfg.setReadThrough(true);
+        cacheCfg.setWriteThrough(true);
```

Comment:
Please remove this line because 'writeThrough' is not support for CacheLoadOnlyStoreAdapter that supports only loading from the underlying storage of file.